src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -330,6 +330,7 @@ protected virtual bool IsAttributeConditionallyOmitted(NamedTypeSymbol attribute
                         }
 
                         BoundExpression boundNamedArgument = BindNamedAttributeArgument(argument, attributeType, diagnostics);
+                        this.Compilation.VariableUsePass.Visit(boundNamedArgument);
```

Comment:
>this.Compilation.VariableUsePass.Visit(boundNamedArgument); [](start = 24, length = 59)

It looks like this code could be executed as part of a speculation, see the following APIs:
- ```public static SymbolInfo GetSpeculativeSymbolInfo(this SemanticModel semanticModel, int position, AttributeSyntax attribute)``` 
- ```public static bool TryGetSpeculativeSemanticModel(this SemanticModel semanticModel, int position, AttributeSyntax attribute, out SemanticModel speculativeModel)```

When that happens, we are not binding code that is actually part of the compilation, we should not be adjusting state in a scenario like this. Please add necessary tests when both mentioned APIs are involved.